e2e/test/scenarios/dashboard-filters/old-parameters.cy.spec.js

```diff
@@ -63,14 +68,35 @@ describe("scenarios > dashboard > OLD parameters", () => {
     it("should work", () => {
       cy.findAllByText("Doohickey");
 
-      cy.contains("Category").click();
+      cy.button("Category").click();
       popover().within(() => {
         cy.findByText("Gadget").click();
         cy.findByText("Add filter").click();
       });
 
       // verify that the filter is applied
       cy.findByText("Doohickey").should("not.exist");
+
+      // test prevent auto-apply filters in slow dashboards metabase#29267
+      openDashboardSidebar();
+
+      const dashboardParameters = cy.findByTestId(
+        "dashboard-parameters-widget-container",
+      );
+
+      dashboardParameters.findByText("Gadget").should("be.visible");
+      rightSidebar().findByLabelText("Auto-apply filters").click();
+      dashboardParameters.button("Apply").should("not.exist");
+
+      cy.button("Category").click();
```

Comment:
off top, but `cy.button` seems to be an antipattern in cypress
<img width="1015" alt="image" src="https://user-images.githubusercontent.com/125459446/233047595-b1f3ed50-cbd5-470a-b1c1-d64f06b84641.png">

Reply:
That excerpt looks a bit different from this case. That one just abstracts `cy.get(selector).click()` but our `cy.button` does help with not having to type long `cy.findByRole("button", { name: buttonName })` every time.

https://github.com/metabase/metabase/blob/34b9ff849b080baeff5858f0aeab7eb35db93b0a/e2e/support/commands/ui/button.js#L1-L3

I think the use case is a bit different, though I personally don't mind typing `.findByRole("button", { name: xxx })` all the time since that's what we do with non-button elements as well. But @nemanjaglumac might be able to give more insight. As he suggests I use `cy.button`.